If an address provider returns null, it produces NPE when establishing connections

Although this exception is then wrapped in the TarantoolConnectionException, there should be more clear exception anyway rather then a rough NPE:

```
Caused by: io.tarantool.driver.exceptions.TarantoolConnectionException: The client is not connected to Tarantool server
	at io.tarantool.driver.core.AbstractTarantoolConnectionManager.lambda$getConnection$0(AbstractTarantoolConnectionManager.java:96) ~[cartridge-driver-0.4.1.jar!/:na]
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture.handle(Unknown Source) ~[na:na]
	at io.tarantool.driver.core.AbstractTarantoolConnectionManager.getConnection(AbstractTarantoolConnectionManager.java:88) ~[cartridge-driver-0.4.1.jar!/:na]
	at io.tarantool.driver.AbstractTarantoolClient.makeRequest(AbstractTarantoolClient.java:411) ~[cartridge-driver-0.4.1.jar!/:na]
	at io.tarantool.driver.AbstractTarantoolClient.call(AbstractTarantoolClient.java:228) ~[cartridge-driver-0.4.1.jar!/:na]
	at io.tarantool.driver.AbstractTarantoolClient.call(AbstractTarantoolClient.java:222) ~[cartridge-driver-0.4.1.jar!/:na]
	at io.tarantool.driver.AbstractTarantoolClient.call(AbstractTarantoolClient.java:216) ~[cartridge-driver-0.4.1.jar!/:na]
	... 33 common frames omitted
Caused by: java.lang.NullPointerException: null
	at io.tarantool.driver.core.AbstractTarantoolConnectionManager.getConnections(AbstractTarantoolConnectionManager.java:175) ~[cartridge-driver-0.4.1.jar!/:na]
	at io.tarantool.driver.core.AbstractTarantoolConnectionManager.establishConnections(AbstractTarantoolConnectionManager.java:160) ~[cartridge-driver-0.4.1.jar!/:na]
	at io.tarantool.driver.core.AbstractTarantoolConnectionManager.getConnectionInternal(AbstractTarantoolConnectionManager.java:112) ~[cartridge-driver-0.4.1.jar!/:na]
	... 38 common frames omitted
```

--- a/src/main/java/io/tarantool/driver/core/connection/AbstractTarantoolConnectionManager.java
+++ b/src/main/java/io/tarantool/driver/core/connection/AbstractTarantoolConnectionManager.java
@@ -177,6 +177,11 @@ public abstract class AbstractTarantoolConnectionManager implements TarantoolCon
 
     private List<CompletableFuture<Map.Entry<TarantoolServerAddress, List<TarantoolConnection>>>> getConnections() {
         Collection<TarantoolServerAddress> addresses = getAddresses();
+        if (addresses == null) {
+            addresses = Collections.emptyList();
+            logger.debug("The list of server addresses is not defined");
+        }
+
         List<CompletableFuture<Map.Entry<TarantoolServerAddress, List<TarantoolConnection>>>> endpointConnections =
                 new ArrayList<>(addresses.size());
         for (TarantoolServerAddress serverAddress : addresses) {

--- a/src/test/java/io/tarantool/driver/integration/ConnectionIT.java
+++ b/src/test/java/io/tarantool/driver/integration/ConnectionIT.java
@@ -1,17 +1,12 @@
 package io.tarantool.driver.integration;
 
+import io.tarantool.driver.api.*;
 import io.tarantool.driver.core.ClusterTarantoolTupleClient;
-import io.tarantool.driver.api.TarantoolClientConfig;
-import io.tarantool.driver.api.TarantoolServerAddress;
-import io.tarantool.driver.api.TarantoolResult;
 import io.tarantool.driver.api.conditions.Conditions;
 import io.tarantool.driver.api.tuple.TarantoolTuple;
 import io.tarantool.driver.auth.SimpleTarantoolCredentials;
 import io.tarantool.driver.auth.TarantoolCredentials;
-import io.tarantool.driver.exceptions.NoAvailableConnectionsException;
-import io.tarantool.driver.exceptions.TarantoolClientException;
-import io.tarantool.driver.exceptions.TarantoolConnectionException;
-import io.tarantool.driver.exceptions.TarantoolInternalException;
+import io.tarantool.driver.exceptions.*;
 import io.tarantool.driver.api.metadata.TarantoolSpaceMetadata;
 import org.junit.jupiter.api.Test;
 import org.slf4j.Logger;
@@ -19,6 +14,7 @@ import org.slf4j.LoggerFactory;
 import org.testcontainers.containers.TarantoolContainer;
 import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.util.ClassUtil;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -321,5 +317,24 @@ public class ConnectionIT {
 
         ClusterTarantoolTupleClient client = new ClusterTarantoolTupleClient(credentials, serverAddress);
         client.close();
+    }
+
+    @Test
+    public void test_AddressProviderReturnsNull_shouldThrowTarantoolClientException() {
+        // given
+        TarantoolClusterAddressProvider addressProvider = () -> null;
+
+        //when
+        TarantoolClient<TarantoolTuple, TarantoolResult<TarantoolTuple>> client = TarantoolClientFactory.createClient()
+                .withAddressProvider(addressProvider)
+                .withCredentials(new SimpleTarantoolCredentials(GUEST_USER, ""))
+                .withConnectTimeout(1000)
+                .withReadTimeout(1000)
+                .withConnections(1)
+                .build();
+
+        // then
+        TarantoolClientException exception =  assertThrows(TarantoolClientException.class, client::getVersion);
+        assertFalse(ClassUtil.getRootCause(exception) instanceof NullPointerException);
     }
 }

